apps/src/gamelab/GameLab.js

```diff
@@ -313,6 +313,11 @@ GameLab.prototype.setupReduxSubscribers = function (store) {
     var lastState = state;
     state = store.getState();
 
+    if (state.interfaceMode !== lastState.interfaceMode && state.interfaceMode === GameLabInterfaceMode.ANIMATION
+      && !this.studioApp_.hasContainedLevels) {
+      this.studioApp_.resetButtonClick();
+    }
```

Comment:
It's possible we're optimizing for a scenario that won't actually be necessary, but I think that if we have a contained level, AND we've submitted an answer (codeStudioLevels.js hasValidContainedLevelResult), we want to behave as normal and do a reset when changing modes.

I also wonder if non-coding tabs will be properly read-only in this scenario (my suspicion is they won't, and this just isn't a fully supported scenario yet)